### PR TITLE
Add plugin functionality

### PR DESCRIPTION
**Pull request by kepoling**

### CODE REVIEW

File: forge/__init__.py [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+class Application:
+    """ Application Class """
     def __init__(self, name):
         self.name = name
 
         self.registry = {}
 
-        self.plugin_source = plugin_base.make_plugin_source(
-            searchpath=['/usr/local/etc/plugins'],
+        self.plugins = []
```
Comment (RAshkettle) [inline forge/__init__.py:21]: All paths should be relative and using the OS library...

import os

        dirname = os.path.dirname(__file__)
        filename = os.path.join(dirname, relative_path_file)

This won't work on Windows at all.  If you don't want a relative path, then the dir must be configurable someplace.

File: forge/__init__.py [hunk 2]
```diff
@@ -29,20 +35,38 @@ def register_plugin(self, name, plugin, helptext):
...
     def execute(self, command, args):
+        """ Execute Plugin """
         if command == 'help':
             self.print_help()
         else:
             self.registry[command][0](args)
 
+    def parse_conf(self, forge_dir, conf_file):
+        """ Parse Plugin Configuration File """
+        config = configparser.ConfigParser()
```
Comment (RAshkettle) [inline forge/__init__.py:53]: Again, look at the os comment above.  This is not only non-portable, but fragile and a bad practice in python.

File: forge/__init__.py [hunk 3]
```diff
@@ -29,20 +35,38 @@ def register_plugin(self, name, plugin, helptext):
...
+
+        if not os.path.exists(forge_dir + conf_file):
+            self._init_conf_file(forge_dir + conf_file, config)
+
+        for key in config['plugin-definitions']:
+            self.plugins.append(forge_dir + 'plugins/' + key)
+
+    @staticmethod
+    def _init_conf_file(path, config):
+        config['plugin-definitions'] = {}
```
Comment (RAshkettle) [inline forge/__init__.py:64]: And you created a resource leak.  Use  **with** to close everything when finished.

File: forge/__init__.py (hunk 2 above)
Comment (RAshkettle) [reply forge/__init__.py:53]: just reading a file doesn't close it.  **with** is your friend here.  Resource leakage.

**Review by RAshkettle — CHANGES_REQUESTED** [review]
Please make the changes below and I'll go over it all again.

File: forge/__init__.py (hunk 1 above)
Comment (kepoling) [reply forge/__init__.py:21]: Gotcha, Ill go with making it configurable for now. I definitely want to support windows installations. Will make a ticket for supporting that.

File: forge/__init__.py (hunk 2 above)
Comment (kepoling) [reply forge/__init__.py:53]: config.read here is just calling the configparser api, I assume internally it handles resource management. When writing to it you'll notice we use `with`.

File: forge/__init__.py (hunk 3 above)
Comment (kepoling) [reply forge/__init__.py:64]: Will do.

**Comment by kepoling** [comment]
Closing to work on my fork, making it more portable and support pypi packaging.

**Pull request closed without merging**